lib/augmentConfig.js

```diff
@@ -275,16 +275,21 @@ function mergeConfigs(a, b) {
 		rulesMerger.rules = { ...a.rules, ...b.rules };
 	}
 
-	const result = {
+	const extensionsMerger = {};
+
+	if (a.extensions || b.extensions) {
+		extensionsMerger.extensions = [...new Set([...(a.extensions || []), ...(b.extensions || [])])];
```

Comment:
[trivial] Using `concat()` seems a bit readable. But just my preference. You can keep it.

```suggestion
		extensionsMerger.extensions = [...new Set((a.extensions || []).concat(b.extensions || []))];
```